### ISSUE TITLE
The METAFONT Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted. ; VSCode

### ISSUE THREAD

**Issue by StratoSTARbyHIRO** [mf2vec-dev/vscode-metafont#1]
The settings with METAFONT and VSCode are default, upon a clean install of VSCode 1.91.1 running on Windows 11. Here is the output from the console window. Further information may be available upon request. 


```
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/AkzidenzGroteskBE-BoldEx.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 4:55:30 PM] Server process exited with code 1.
[Info  - 4:55:30 PM] Connection to server got closed. Server will restart.
true
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/AkzidenzGroteskBE-BoldEx.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 4:55:30 PM] Server process exited with code 1.
[Info  - 4:55:30 PM] Connection to server got closed. Server will restart.
true
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/AkzidenzGroteskBE-Super.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 4:55:30 PM] Server process exited with code 1.
[Info  - 4:55:30 PM] Connection to server got closed. Server will restart.
true
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/AkzidenzGroteskBE-BoldEx.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 4:55:31 PM] Server process exited with code 1.
[Info  - 4:55:31 PM] Connection to server got closed. Server will restart.
true
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/AkzidenzGroteskBE-BoldEx.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 4:55:31 PM] Server process exited with code 1.
[Error - 4:55:31 PM] The METAFONT Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

**Comment by mf2vec-dev** [comment]
Thanks for the bug report, @DDroid.

According to the error message there are (at least) two `.mf` files in your workspace that VSCode identifies as binary (e.g. `C:\Users\joshu\Desktop\pdb\AkzidenzGroteskBE-BoldEx.mf` and `AkzidenzGroteskBE-Super.mf`).
Can you please check whether these are really binary files?
(E.g. open the file in VSCode. If there is a warning, click "Open Anyway" and select "Text Editor" if there is a follow-up menu. If the file contains mostly random symbols, the file you have opened is a binary file.)

If the files are binary:
Could you please change the file extension of all these binary files, e.g. rename `AkzidenzGroteskBE-BoldEx.mf` to `AkzidenzGroteskBE-BoldEx.mf_temp`. Please restart VSCode and check if the error still occurs. You may get more error messages with other files, rename the error causing files until the error messages are different or there are no more errors.

If the files are _not_ binary, e.g. valid METAFONT code:
It would be super helpful if you could upload one of the files (e.g. `AkzidenzGroteskBE-BoldEx.mf`) so that I can reproduce the problem to trace the issue.

**Comment by StratoSTARbyHIRO** [comment]
> Thanks for the bug report, @StratoSTARbyHIRO 
> 
> According to the error message there are (at least) two `.mf` files in your workspace that VSCode identifies as binary (e.g. `C:\Users\joshu\Desktop\pdb\AkzidenzGroteskBE-BoldEx.mf` and `AkzidenzGroteskBE-Super.mf`). Can you please check whether these are really binary files? (E.g. open the file in VSCode. If there is a warning, click "Open Anyway" and select "Text Editor" if there is a follow-up menu. If the file contains mostly random symbols, the file you have opened is a binary file.)
> 
> If the files are binary: Could you please change the file extension of all these binary files, e.g. rename `AkzidenzGroteskBE-BoldEx.mf` to `AkzidenzGroteskBE-BoldEx.mf_temp`. Please restart VSCode and check if the error still occurs. You may get more error messages with other files, rename the error causing files until the error messages are different or there are no more errors.
> 
> If the files are _not_ binary, e.g. valid METAFONT code: It would be super helpful if you could upload one of the files (e.g. `AkzidenzGroteskBE-BoldEx.mf`) so that I can reproduce the problem to trace the issue.

Hello, thank you for your reply @mf2vec-dev 

 It seems like all the .mf files are indeed binary as they look like a bunch of jumbled up symbols both in VSCode, and in a separate Notepad++. Upon renaming each of the offending `.mf` file to `.mf_temp` in stages, I am left with no more readable `.mf` files (as they are now all `.mf_temp`, due to each one causing what appears to be the same error.) I took only one `.mf` file at a time in a separate directory and the issue still persists. 

```
Node.js v20.9.0
[Error - 3:25:17 PM] Server process exited with code 1.
[Info  - 3:25:17 PM] Connection to server got closed. Server will restart.
true
c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6
${JSON.stringify(y.error,void 0,4)}`):a.error("Received response message without id. No further error information provided.");else{let L=y.id,K=z.get(L);if(It(y,K),K!==void 0){z.delete(L);try{if(y.error){let Q=y.error;K.reject(new ae.ResponseError(Q.code,Q.message,Q.data))}else if(y.result!==void 0)K.resolve(y.result);else throw new Error("Should never happen.")}catch(Q){Q.message?a.error(`Response handler '${K.method}' failed with message: ${Q.message}`):a.error(`Response handler '${K.method}' failed unexpectedly.`)}}}}function Pn(y){if(Jt())return;let L,K;if(y.method===To.type.method){let Q=y.params.id;ce.delete(Q),Un(y);return}else{let Q=$.get(y.method);Q&&(K=Q.handler,L=Q.type)}if(K||E)try{if(Un(y),K)if(y.params===void 0)L!==void 0&&L.numberOfParams!==0&&L.parameterStructures!==ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received none.`),K();else if(Array.isArray(y.params)){let Q=y.params;y.method===bo.type.method&&Q.length===2&&Jc.is(Q[0])?K({token:Q[0],value:Q[1]}):(L!==void 0&&(L.parameterStructures===ae.ParameterStructures.byName&&a.error(`Notification ${y.method} defines parameters by name but received parameters by position`),L.numberOfParams!==y.params.length&&a.error(`Notification ${y.method} defines ${L.numberOfParams} params but received ${Q.length} arguments`)),K(...Q))}else L!==void 0&&L.parameterStructures===ae.ParameterStructures.byPosition&&a.error(`Notification ${y.method} defines parameters by position but received parameters by name`),K(y.params);else E&&E(y.method,y.params)}catch(Q){Q.message?a.error(`Notification handler '${y.method}' failed with message: ${Q.message}`):a.error(`Notification handler '${y.method}' failed unexpectedly.`)}else ut.fire(y)}function Bn(y){if(!y){a.error("Received empty message.");return}a.error(`Received message which is neither a response nor a notification message:
                                                                                                                                                                                                                                   ^

mo [Error]: Request OpenAllFilesRequest failed with message: cannot open file:///c%3A/Users/joshu/Desktop/pdb/mf_basetest/Helvetica.mf. Detail: File seems to be binary and cannot be opened as text
    at Pt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:6:228)
    at br (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6260)
    at Bt (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6385)
    at Immediate.<anonymous> (c:\Users\joshu\.vscode\extensions\mf2vec-dev.vscode-metafont-0.3.1\out\language\server.js:5:6147)
    at process.processImmediate (node:internal/timers:478:21) {
  code: -32603,
  data: undefined
}

Node.js v20.9.0
[Error - 3:25:18 PM] Server process exited with code 1.
[Error - 3:25:18 PM] The METAFONT Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
``` 

I assume since the error is all the same for each `.mf` file, it doesn't matter which file I send so I will send you this one in the latest error "`Helvetica.mf`" in ZIP format due to GitHub's file upload not recognizing `.mf` files. [Helvetica_mf.zip](https://github.com/user-attachments/files/16395998/Helvetica_mf.zip)

**Comment by mf2vec-dev** [comment]
OK. Thank you for providing the information and the file. I think this is indeed a bug in `vscode-metafont` as the language server should not crash. Fixing this would only skip binary files, so you would still not be able to use these files. Nevertheless, I'll include a fix in the next update.

The file in the zip you uploaded is indeed binary. I couldn't determine the file type. It doesn't seem to be a common font format, nor a METAFONT base file. Also, `mf` cannot read the file.

Where did you get the file from (or the binary .mf files in general)?

If you want to run METAFONT, you need files written in the METAFONT language (`vscode-metafont` aims to help you with this). You can find many introductory examples online ([this](https://www.win.tue.nl/~aeb/tex/mf/metafont.html) is an introduction by prof.dr. A.E. Brouwer). I strongly recommend that you take a look at The METAFONTbook or other tutorials if you are not familiar with the METAFONT language or the `mf` program. The METAFONTbook can be found in many libraries. You should also be able to find a PDF version of The METAFONTbook using your favorite web search engine.

**Comment by StratoSTARbyHIRO** [comment]
Thank you for your response! 

Understood, and the `.mf` files I have came from an old Silicone Graphics IRIX system, it seems like everything about them is some horrible proprietary mess. So while they seem to use the METAFONT `.mf` file extension, they must evidently be IRIX proprietary instead. Other extensions found there included `.of` and `.sf` as well as `.mf`, none of which so far I have managed to find a way to open/use/convert.

Thank you for your help however, I deffo appreciate it!
